lua/apisix/core/request.lua

```diff
@@ -19,6 +22,27 @@ local function _headers(ctx)
 
     return headers
 end
+
+local function _validate_header_name(name)
+    local tname = type(name)
+    if tname ~= "string" then
+        return nil, str_fmt("invalid header name %q: got %s, " ..
+                "expected string", name, tname)
+    end
+
+    return name, nil
```

Comment:
`return name` is enough.

Reply:
return only the `name` or cause the use case `t/node/invalid-service.t` to throw an exception.

Reply:
I think it's nonbusiness with `return name`. I will try to dig.

Reply:
Yes, it's not related to the name of the return parameter. What I want to express is related to the number of parameters.

Reply:
it's run ok on my side after remove useless `nil`.

Reply:
fixed.